react/features/overlay/components/OverlayContainer.js

```diff
@@ -0,0 +1,202 @@
+/* global APP */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PageReloadOverlay from './PageReloadOverlay';
+import SuspendedOverlay from './SuspendedOverlay';
+import UserMediaPermissionsOverlay from './UserMediaPermissionsOverlay';
+
+/**
+ * Implements a React Component that will display the correct overlay when
+ * needed.
+ */
+class OverlayContainer extends Component {
+    /**
+     * OverlayContainer component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The browser which is used currently.
+         * NOTE: Used by UserMediaPermissionsOverlay only.
+         * @private
+         * @type {string}
+         */
+        _browser: React.PropTypes.string,
+
+        /**
+         * The indicator which determines whether the status of
+         * JitsiConnection object has been "established" or not.
+         * NOTE: Used by PageReloadOverlay only.
+         * @private
+         * @type {boolean}
+         */
+        _connectionEstablished: React.PropTypes.bool,
+
+        /**
+         * The indicator which determines whether a critical error for reload
+         * has been received.
+         * NOTE: Used by PageReloadOverlay only.
+         * @private
+         * @type {boolean}
+         */
+        _haveToReload: React.PropTypes.bool,
+
+        /**
+         * The indicator which determines whether the reload was caused by
+         * network failure.
+         * NOTE: Used by PageReloadOverlay only.
+         * @private
+         * @type {boolean}
+         */
+        _isNetworkFailure: React.PropTypes.bool,
+
+        /**
+         * The indicator which determines whether the GUM permissions prompt
+         * is displayed or not.
+         * NOTE: Used by UserMediaPermissionsOverlay only.
+         * @private
+         * @type {boolean}
+         */
+        _mediaPermissionPromptVisible: React.PropTypes.bool,
+
+        /**
+         * The reason for the error that will cause the reload.
+         * NOTE: Used by PageReloadOverlay only.
+         * @private
+         * @type {string}
+         */
+        _reason: React.PropTypes.string,
+
+        /**
+         * The indicator which determines whether the GUM permissions prompt
+         * is displayed or not.
+         * NOTE: Used by SuspendedOverlay only.
+         * @private
+         * @type {string}
+         */
+        _suspendDetected: React.PropTypes.bool
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement|null}
+     * @public
+     */
+    render() {
+        // FIXME: Temporary workaround until everything is moved to react.
+        APP.UI.overlayVisible
```

Comment:
if we decide to go with a class component (see above) this can be done in lifecycle methods, which is kinda less hacky.